### DISCUSSION TITLE
Time to move on to ReForge or to A1111's WebUIs

### DISCUSSION THREAD

**Discussion by Bullseye110** [lllyasviel/stable-diffusion-webui-forge#900] [General]
Hello!
Forge was a very good WebUI, but it seems that illyasviel is doing what he said a while ago with this project: experimenting.
This way, Forge is now very much unusable especially on cloud GPUs and pretty much all the older extensions won't work either.
It was the best repo for running SD (UI) out there, but yea, now we have to move on to ReForge or to the classic A1111 WebUI.

Anyway, for the people who still wish to use Forge (the stable and working one), just use the commit "a9e0c38" (from the 22nd of July, 2024). Everything after that date just broke Forge.
This is how to do it:
git clone https://github.com/lllyasviel/stable-diffusion-webui-forge
cd C:\Users\Admin\stable-diffusion-webui-forge
git checkout a9e0c38

Good luck and peace!
Too bad that this project is done for almost all the cloud GPUs users.

**Comment by starbez** [comment]
Ever since the update, Forge has become less responsive for me.

**Reply by Bullseye110** [reply]
Exactly that.

**Comment by CCpt5** [comment]
You should do what you want to do, but kinda lame to post on this repo with this sort of comment.

Also read the update in Readme from the other day - @lllyasviel explains there are major revisions ongoing but he expects them to be done on the 7th - https://github.com/lllyasviel/stable-diffusion-webui-forge/commit/5f9e0473010d9a9a9649aefac6a9d768098c1355

Walk where you like your steps - lots of options.

**Reply by Bullseye110** [reply]
I wrote that for the users who legit used and found this Forge project very useful being the best alternative to A1111 for low end PCs.

**Reply by starbez** [reply]
> I wrote that for the users who legit used and found this Forge project very useful being the best alternative to A1111 for low end PCs.

 turning off [sd-webui-civbrowser ](https://github.com/SignalFlagZ/sd-webui-civbrowser)made the UI faster for me. you should try this.

**Reply by Strapples** [reply]
> > I wrote that for the users who legit used and found this Forge project very useful being the best alternative to A1111 for low end PCs.
> 
> turning off [sd-webui-civbrowser ](https://github.com/SignalFlagZ/sd-webui-civbrowser)made the UI faster for me. you should try this.

thanks, always looking for ways to prune off extras I don't need. Running mine on a 4GB laptop and it chugs along fine. It's not fast but I'm making it generate 1080p native... because I'm a cruel $&@! to graphics cards.

I swear when the AI's become sentient the graphics cards are going to come to kill us all.

**Reply by Panchovix** [reply]
To add, on reForge I'm trying to add support for new models but failing (this was way harder than expected for me, patching the forge loader is not joke) and OG forge after the backend will for sure support sd3 and flux. I still can't add either of those. 

A1111 has more support but performance is an important part where forge can give better performance in some cases.

So I still think Forge is the future, but it will take a bit of time.

**Comment by Strapples** [comment]
Feel this is my best chance at making my voice heard to the head developers of this project.

WHAT ARE YOU THINKING!?

This is a great project, don't turn your main branch into your dev branch and muck it up for everyone just so you can code faster.

Why can't you create a dev branch? 

Is it too hard to maintain a mainline branch (stable) and then have your canary builds (dev?) cause Firefox been doing it since they were two guys.

Dirty fix for now, if you're still running a June build, delete your updater package and never update the thing *EVER*

Maybe the devs will post on here or somewhere when they come across a version thats good enough to call a "Gold Master Forge Update" or "Forge LTS"

For now I'm running the June 27th 2024 build and will keep running it.

A lot of people in Linux don't get their updates for a year or two while the dev boys get their stuff every month or weekly depending on what channel they're on (or nightly, if they really wanna blow their computer up)

Let's just sit back and keep working with what we have, Forge as it stands June 27th is still a freaking powerhouse of a system I have yet to scratch the surface of. There's so many pieces of tech to manipulate and get art from.

Hopefully the developers figure out a way to retain existing Forge users who run production environments that cannot be safely migrated without data loss.

**Comment by CCpt5** [comment]
**Current Progress (2024 Aug 3):** Backend Rewrite is 80% finished - remaining 30 hours to begin making it stable; remaining 48 hours to begin supporting many new things.

https://github.com/lllyasviel/stable-diffusion-webui-forge/commit/0b129b71811ee470695a3cd88c249fb7690cf8e2

**Reply by Bullseye110** [reply]
Hopefully this will be true.

**Reply by Strapples** [reply]
Sooo basically a 3 - 6 month stable branch schedule. Nice.

**Comment by Bullseye110** [comment]
Update: 12.08.2024

So I tried the latest Forge (which seems to be 95.5% completed) and it's so disappointing that you can't believe how much down the road this project went.
I tested it on a cloud GPU with 16 GB of VRAM.
1. The user interface is a complete joke. It feels like you are playing CS GO with 5 FPS and 300 ping. That's legit how laggy it feels.
2. You press the generate button and wait one minute before actually seeing the image starting generating (I mean come on, even the base A1111 is faster...).
3. Not all the ControlNets are actually working and it's a pain to swap the preprocessors/cnet models as you have to start generating -> stop -> start generating again for the controlnet to even have effect. Union Promax Canny is working, not a single openpose model is working.
4. 90% of the extensions are broken, at least aDetailer is working.

You will be wondering, but the actual A1111 is better than this Forge in this state. Unbelievable...

Edit: And no, it's not the cloud GPU provider's fault. I tested all of the WebUIs (A1111, Forge, ReForge) and only the newest Forge has these many problems.

**Reply by CCpt5** [reply]
Yea, lllyasviel has been coding like a madman the past week to get Forge up to state of the art and to support Flux. If you don't follow his commits/updates in Readme then yes it's best you go back to A1111 or Reforge for a few days to a few weeks until he gets everything streamlined.

Not sure how connected you are with what's going on in the community, but "Flux" came out on August 1 and is significantly more resource heavy then prior models. Results are much better also. That is why generating images might be taking a lot longer than before.

This is a free project so really should consider that to not discourage the developer when posting. Many of us are ecstatic he even revisited this project after it became seemingly abandoned for 3+ months.

Good luck.

His commits/code updates:
https://github.com/lllyasviel/stable-diffusion-webui-forge/commits/main/

**Reply by ishadowx27** [reply]
Well, I have a crappy PC with a 1070 8GB GPU and the latest commit  as of this moment works beautifully, even with flux giving me 2-4 steps astounding 1216px generations in less than a minute.  So yeah, it is you and your "cloud" service. If you are disappointed by a tool and its developer even though you give nothing to support it other than annoyingly crying like a baby online, maybe you should try paid services.

**Reply by yamfun** [reply]
we are all using the new Forge happily for Flux, must be your issue

**Reply by Bullseye110** [reply]
> we are all using the new Forge happily for Flux, must be your issue

Do you use any other extension excepting the image generation with Flux? I thought so...

**Comment by Strapples** [comment]
Hey guys! I got a wonderful idea! It’s called stop updating it constantly. You. You. You are all hobbyists. Artists usually don’t need the cutting edge release for everything. These monthly builds. They’re basically chrome canary. You wanna follow this project then you’re using Auto1111 Canary builds. There will be stable canary builds. I was watching someone who said that this would have a build that should be considered as a LTS candidate in September or October. We need to treat this project like a canary program that releases an LTS about twice to three times a year. You can go bleeding edge or take the LTS build. I’ll just stick to LTS releases. Sent from my iPhoneOn Aug 12, 2024, at 5:48 PM, CCpt5 ***@***.***> wrote:﻿
Yea, lllyasviel has been coding like a madman the past week to get Forge up to state of the art and to support Flux. If you don't follow his commits/updates in Readme then yes it's best you go back to A1111 or Reforge for a few days to a few weeks until he gets everything streamlined.
Not sure how connected you are with what's going on in the community, but "Flux" came out on August 1 and is significantly more resource heavy then prior models. Results are much better also. That is why generating images might be taking a lot longer than before.
This is a free project so really should consider that to not discourage the developer when posting. Many of us are ecstatic he even revisited this project after it became seemingly abandoned for 3+ months.
Good luck.
His commits/code updates:
https://github.com/lllyasviel/stable-diffusion-webui-forge/commits/main/

**Comment by Strapples** [comment]
Yo shadow can you tell me your build number so I can grab that build?Sent from my iPhoneOn Aug 12, 2024, at 7:57 PM, ishadowx27 ***@***.***> wrote:﻿
Well, I have a crappy PC with a 1070 8GB GPU and the latest commit  as of this moment works beautifully, even with flux giving me 2-4 steps astounding 1216px generations in less than a minute.  So yeah, it is you and your "cloud" service. If you are disappointed by a tool and its developer even though you give nothing to support it other than annoyingly crying like a baby online, maybe you should try paid services.

**Reply by ishadowx27** [reply]
3589b57 is the one I was using at the time but I updated and things haven't changed much. The latest build now supports LORAs and is a few seconds slower for me and my lame videocard but I am not in a rush for my generations (since they are purely personal so far)

**Comment by Strapples** [comment]
What build number y’all marking as stable? so I can get the best experience?Sent from my iPhoneOn Aug 13, 2024, at 9:42 AM, yamfun ***@***.***> wrote:﻿
we are all using the new Forge happily for Flux, must be your issue

**Comment by SillySilk** [comment]
I think most of us are pretty happy with what's been done so far and we
understand that it is a work in progress and I can say for myself I'm
utterly thrilled because I can't stand comfy and Would like to avoid using
it if at all possible.
Right now these are very early days and complaining that something that is
being offered to you for free which is currently being worked on is not up
to your standards, I would just say there's the door dude.

**Comment by ali0une** [comment]
LOL nope ... our Lord and Savior @lllyasviel is back, rejoice Forge children!

**Comment by Strapples** [comment]
Thanks shadow. Sent from my iPhoneOn Aug 14, 2024, at 4:39 AM, ishadowx27 ***@***.***> wrote:﻿
3589b57 is the one I was using at the time but I updated and things haven't changed much. The latest build now supports LORAs and is a few seconds slower for me and my lame videocard but I am not in a rush for my generations (since they are purely personal so far)

**Comment by SuperChevyChevy** [comment]
Sorry, Im not experienced with Git, im not sure how to follow the instructions to go to the older version. Can someone assist with what i need to do and where? This newest update has broken the refiner option, which i need for work.

**Reply by Mol31** [reply]
You can download previous stable version here https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/previous/webui_forge_cu121_torch21_f0017.7z

**Reply by Strapples** [reply]
> You can download previous stable version here https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/previous/webui_forge_cu121_torch21_f0017.7z

Ty, that’s the current long term stable I see. 

I see forge coming with another LTS in Sept or October

**Reply by SuperChevyChevy** [reply]
> You can download previous stable version here https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/previous/webui_forge_cu121_torch21_f0017.7z

So all I need to do is follow the link, and download and install like normal? It won’t auto update or anything?

**Reply by Bullseye110** [reply]
> > You can download previous stable version here https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/previous/webui_forge_cu121_torch21_f0017.7z
> 
> So all I need to do is follow the link, and download and install like normal? It won’t auto update or anything?

Nope if you don't run the "update.bat".

**Comment by 100xlongx** [comment]
this guy illyasviel is the goat. man is building and delivering so much. perhaps just use a different ui instead of demoralizing people.

**Comment by ghost** [comment]
Sure, experimenting is good and Illyasviel has to be really talented with all of this stuff. But the management of the project is... not as good as the actual technical skills. You know, all the experimentation could be done in an experimental or development branch. The one you checkout into at your own peril. And many people love that. They experiment with the cutting edge features and report any issues they find. But for those of us who want stability, this has made forge completely unusable. 

For now I switched completely to reforge only because it has some of the latest features that forge didn't have in May (the last stable version) but nothing is broken, at least so badly. 

My advise is: experimental stuff in its own branch. There's some extra work, especially with making sure merge conflicts against master don't go out of hand, if master gets other updates, but still. You can't have a "95% rework done" thing in the master branch. Because one branch is needed to be in a state that it works - I mean, there are always bugs and issues, but in a stable state.

**Reply by Bullseye110** [reply]
I said kinda the same things, but these people only know to dislike the posts.
It's just pure logic to have 2 branches: one for stability and the other one for experimentation.
90% of them are just happy that they can use Flux, but they don't really care at what cost. I'll tell you at what cost. At the cost of slowing down all the XL generations/hires. fixes. and at the cost of making 90% of the extensions unusable.
This is legit a try for "innovation" without foundation. Build a house on a very bad foundation and "innovate" it with 30 rooms. At the first earthquake, the house will be reduced to dust no matter how many fancy rooms you have.

**Reply by alienzed** [reply]
I think the downvotes come from people sounding entitled... There's a way to say "pretty please" all while saying "thank you so much" at the same time that won't get the downvotes :)

**Reply by Dahvikiin** [reply]
I think the dislikes come from people who can read and understand when Illyasviel says "This repository is for experimentation", "Forge is an experiment" and therefore the experimental, dev, beta, etc... branches are unnecessary, because **STABLE doesn't exist**, **FORGE=EXPERIMENTAL**. and at best you don't pull the changes.

**Reply by alienzed** [reply]
Makes sense... with that said... he created something that was stable and people started relying on it. Whether it was his intention or not, he created a market for something where none existed prior (or at least, created a product in a market that people loved).

Then... one might assume that executing "git pull", without changing branches, wouldn't wreck things quite this way... I mean, would I be wrong for thinking that based on how most repos work?

The fact is that that's exactly what I did, and it was how I discovered Flux, believe it or not, just by executing git pull. So take this as a feature request, a stable branch :P

He's free to ignore it, but we're not wrong to ask for it, right? People certainly shouldn't be complaining though.

**Reply by vonespy** [reply]
@alienzed I agree. This is a free project put together for us by one of the premier SD developers on this planet. "Please consider doing X" is different from "dev is a bad person and my world is ending because he changed something".

As always, [there's an XKCD for this.](https://xkcd.com/1172/)

I've seen way too many talented people at the top of projects like this quit because they get overwhelmed by the complainers and their entitlement. Or become angry, cynical hermits who don't listen to anyone anymore. Some people can develop a filter for that kind of negativity, and some can't. Nobody should have to though, and I'd hate for lllyasviel to drop what he's doing because the thrill of coding incredible stuff doesn't overcome the aggravation from the community.

**Comment by Strapples** [comment]
Then use an old stable version. Long Term Support releases aren’t every month on Linux. Sent from my iPhoneOn Aug 15, 2024, at 11:04 AM, culturedniichan ***@***.***> wrote:﻿
Sure, experimenting is good and Illyasviel has to be really talented with all of this stuff. But the management of the project is... not as good as the actual technical skills. You know, all the experimentation could be done in an experimental or development branch. The one you checkout into at your own peril. And many people love that. They experiment with the cutting edge features and report any issues they find. But for those of us who want stability, this has made forge completely unusable.
For now I switched completely to reforge only because it has some of the latest features that forge didn't have in May (the last stable version) but nothing is broken, at least so badly.
My advise is: experimental stuff in its own branch. There's some extra work, especially with making sure merge conflicts against master don't go out of hand, if master gets other updates, but still. You can't have a "95% rework done" thing in the master branch. Because one branch is needed to be in a state that it works - I mean, there are always bugs and issues, but in a stable state.

**Comment by vonespy** [comment]
This is peak main character syndrome entitlement. 

You have three choices here:

* Open up your checkbook and start writing huge monthly checks to Illyasviel to compensate him for working on what you want rather than what he feels is valuable to the whole community.
* Branch the code, open up a couple of coding books and figure out how to do this yourself.
* Shut up.

**Reply by alienzed** [reply]
Option 2 is here:

https://github.com/Panchovix/stable-diffusion-webui-reForge

**Reply by vonespy** [reply]
Neat! I somehow missed that. @Bullseye110, could you edit your original post to include info about ReForge?

**Reply by Bullseye110** [reply]
> Neat! I somehow missed that. @Bullseye110, could you edit your original post to include info about ReForge?

I already mentioned ReForge, are you blind or? And I say these things, because I care about this project. If I wouldn't, I will just go to use a1111/ReForge and gg and won't waste time writing the main errors/problems here.

**Reply by alienzed** [reply]
Chill man, he started by saying "I somehow missed that." Not sure "are you blind?" is warranted here... what if he IS blind? Blind people are people too ;)

**Reply by Bullseye110** [reply]
> Chill man, he started by saying "I somehow missed that." Not sure "are you blind?" is warranted here... what if he IS blind? Blind people are people too ;)

Yea, man, I know, but you just can't with this kind of people...

**Reply by alienzed** [reply]
blind people? ;)

they don't see what I did there lol

**Reply by vonespy** [reply]
> > Chill man, he started by saying "I somehow missed that." Not sure "are you blind?" is warranted here... what if he IS blind? Blind people are people too ;)
> 
> Yea, man, I know, but you just can't with this kind of people...

... what kind of people?

**Reply by Bullseye110** [reply]
> > > Chill man, he started by saying "I somehow missed that." Not sure "are you blind?" is warranted here... what if he IS blind? Blind people are people too ;)
> > 
> > 
> > Yea, man, I know, but you just can't with this kind of people...
> 
> ... what kind of people?

Your kind. You take everything as hate. I just pointed out some new problems which arrived with all this "experimental" stuff after the 21st of July 2024. More people talked about these.
I respect Illyasviel, but I'll point out when such problems appear.
Imagine that the new Forge is now even slower than the simple A1111. And the list can continue.
The people are somehow "happy" with the newest forge only because it gives them the chance to play a little bit with some flux models (and flux as a base model is very hyped nowadays).

**Reply by vonespy** [reply]
> > > > Chill man, he started by saying "I somehow missed that." Not sure "are you blind?" is warranted here... what if he IS blind? Blind people are people too ;)
> > > 
> > > 
> > > Yea, man, I know, but you just can't with this kind of people...
> > 
> > 
> > ... what kind of people?
> 
> Your kind. You take everything as hate. I just pointed out some new problems which arrived with all this "experimental" stuff after the 21st of July 2024. More people talked about these. I respect Illyasviel, but I'll point out when such problems appear. Imagine that the new Forge is now even slower than the simple A1111. And the list can continue. The people are somehow "happy" with the newest forge only because it gives them the chance to play a little bit with some flux models (and flux as a base model is very hyped nowadays).

I asked you once before and you didn't really tell me much other than I take everything as hate. But then you seemed to get distracted by petty grievances with someone who gives his time to you and the AI art community for free.

So again. What kind of people?
